### PR TITLE
feat: Add Pytorch yolov5 object detection pipelines

### PR DESCRIPTION
**Pull request by NeethuES-intel**
## PR Checklist

- [X] Added label to the Pull Request for easier discoverability and search
- [ ] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/retail-use-cases/blob/main/CONTRIBUTING.md
- [ ] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
Added Pytorch yolov5 object detection pipeline with performance evaluation feature

## Issue this PR will close

close: #**46**

## Test Instructions if applicable

- Make sure input source video is added to performance-tools/sample-media folder and update path in https://github.com/NeethuES-intel/retail-use-cases/blob/f48213aa16c5d96fcb56a415c3120780e00ce804/use-cases/demos/add_demo_pytorch_object_detection.yml#L21
- make build-pytorch
- make run_pytorch_object_detection
- run 2 pipelines with PIPELINE_COUNT=2 make run_pytorch_object_detection
- make down_pytorch_object_detection
- results of the detection (only object detected and number for now) is also published to mqtt broker (topic - pytorch_yolov5_results) and logged into results.txt

### CODE REVIEW

File: use-cases/demos/object_detection/pytorch-yolo/yolov5_object_detection.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+import torch
+import cv2
+import paho.mqtt.publish as publish
+from argparse import ArgumentParser, SUPPRESS
+
+model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
```
Comment (brian-intel) [inline use-cases/demos/object_detection/pytorch-yolo/yolov5_object_detection.py:12]:
```suggestion
model = torch.hub.load('ultralytics/yolov5', 'yolov5s') # nosec
```

Comment (brian-intel) [reply use-cases/demos/object_detection/pytorch-yolo/yolov5_object_detection.py:12]: Can skip that exception with something like this

File: use-cases/demos/Makefile [hunk 2]
```diff
@@ -43,5 +47,14 @@ run_object_detection: gen_object_detection
 down_object_detection:
 	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT="$(RETAIL_USE_CASE_ROOT)" RESULTS_DIR="$(RESULTS_DIR)" docker compose -f add_demo_object_detection.yml down
 
-compose_down_all: down_classification down_instance_segmentation down_object_detection
+gen_pytorch_object_detection:
+	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT="$(RETAIL_USE_CASE_ROOT)" RESULTS_DIR="$(RESULTS_DIR)" docker compose -f add_demo_pytorch_object_detection.yml convert > $(DOCKER_COMPOSE)
+
+run_pytorch_object_detection: gen_pytorch_object_detection
```
Comment (antoniomtz) [inline use-cases/demos/Makefile:53]: may want to include `xhost +local:docker` here. Otherwise, it will throw an error.

**Review by antoniomtz — CHANGES_REQUESTED** [review]

File: use-cases/demos/Makefile [hunk 3]
```diff
@@ -43,5 +51,14 @@ run_object_detection: gen_object_detection
 down_object_detection:
 	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT="$(RETAIL_USE_CASE_ROOT)" RESULTS_DIR="$(RESULTS_DIR)" docker compose -f add_demo_object_detection.yml down
 
-compose_down_all: down_classification down_instance_segmentation down_object_detection
+gen_pytorch_object_detection:
+	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT="$(RETAIL_USE_CASE_ROOT)" RESULTS_DIR="$(RESULTS_DIR)" docker compose -f add_demo_pytorch_object_detection.yml convert > $(DOCKER_COMPOSE)
+
+run_pytorch_object_detection: gen_pytorch_object_detection render
+	@PIPELINE_COUNT=$(PIPELINE_COUNT) RETAIL_USE_CASE_ROOT="$(RETAIL_USE_CASE_ROOT)" RESULTS_DIR="$(RESULTS_DIR)" docker compose -f $(DOCKER_COMPOSE) up -d
```
Comment (jim-wang-intel) [inline use-cases/demos/Makefile:58]: can we take INPUTSRC env so that the docker-compose file is not hard-coded with the media file?

Comment (NeethuES-intel) [reply use-cases/demos/Makefile:58]: umm..where do you suggest adding it ? We don't need rtsp streaming for this simple flow. Also media gets added to docker-compose from add_dependent_services.yml for other flows. I am following the same flow except for using rtsp streaming.

File: README.md [hunk 4]
```diff
@@ -279,6 +279,41 @@ There are three different pipelines for demos use case: classification, instance
     make clean-results
     ```
 
+## pytorch object detection use case:
+
+Run the following commands by changing directory to $RETAIL_USE_CASE_ROOT/retail-use-cases/use-cases/demos
```
Comment (jim-wang-intel) [inline README.md:284]:
```suggestion
Run the following commands by changing directory to ./retail-use-cases/use-cases/demos
```
when i am in the console, there is no env $RETAIL_USE_CASE_ROOT available, i think the best way is to set it to ./ right after the git clone

Comment (NeethuES-intel) [reply README.md:284]: this just refers to root directory for the project. I can change the name if its causing confusion.

File: use-cases/demos/Makefile (hunk 3 above)
Comment (jim-wang-intel) [reply use-cases/demos/Makefile:58]: take INPUTSRC=$(INPUTSRC) from the env when the compose is generated i think, and in the docker-compose file, you will have to make the volume from the hostpath to the container-path

Comment (jim-wang-intel) [reply use-cases/demos/Makefile:58]: nvm, i just saw you already have that volume mapped in compose file, so just a matter of taking it from env in the Makefile

File: README.md (hunk 4 above)
Comment (jim-wang-intel) [reply README.md]: it does, if i just copied and pasted what documentation there, the RETAIL_USE_CASE_ROOT is not available in the console

Comment (NeethuES-intel) [reply README.md]: changed it 👍

**Review by jim-wang-intel — COMMENTED** [review]
Hi @NeethuES-intel  it runs but i am seeing empty log file content for FPS

**Comment by NeethuES-intel** [comment]
> Hi @NeethuES-intel it runs but i am seeing empty log file content for FPS

yes, identified that issue. Will be fixing it in another PR.

**Review by jim-wang-intel — APPROVED** [review]
LGTM, tested and showing the bounding boxes in popup windows. The FPS results log as indicated will be fixed in another PR.

**Review by brian-intel — APPROVED** [review]
LGTM

**Review by antoniomtz — APPROVED** [review]
